include/radproto/socket.h

```diff
@@ -0,0 +1,31 @@
+#ifndef SOCKET_H
```

Comment:
`SOCKET_H` is rather common. Either add a library prefix (`RADPROTO_SOCKET_H`), or use more modern `#pragma once` instead of `#ifndef`/`#define`/`#endif`.

Make sure you do it in every header file.